### DISCUSSION TITLE
How to use store's state inside that store itself?

### DISCUSSION THREAD

**Discussion by enzadevs** [pmndrs/zustand#2105] [Q&A] [answered]
I will be just replicating my whole zustand store in simple manner like this:
```javascript
export const usePrintStore = create((set) =>({
    inputFieldTextValue: '',
    updateInputFieldTextValue: inputFieldTextValue => set(() => ({ inputFieldTextValue: inputFieldTextValue })),
    printToConsole: () => {
        console.log(inputFieldTextValue)
    }
}))
``` 
And I get this error when i call printToConsole():
```javascript
ReferenceError: inputFieldTextValue is not defined
``` 
I can easily change the value for inputFieldTextValue with updateInputFieldTextValue, no problems here. But I need a way to get state's value inside zustand store itself or, maybe, another way out of that problem. I want to print out current value of inputFieldTextValue state on button click.
I'm using zustand version 4.4.3.
Thanks

**Comment by dai-shi** [comment, marked as answer]
Here you go:
```js
export const usePrintStore = create((set, get) =>({
    inputFieldTextValue: '',
    updateInputFieldTextValue: inputFieldTextValue => set(() => ({ inputFieldTextValue: inputFieldTextValue })),
    printToConsole: () => {
        console.log(get().inputFieldTextValue)
    }
}))
```

**Reply by enzadevs** [reply]
Thank you very much! Exactly what I needed.
